crates/fiber/src/unix.rs

```diff
@@ -75,37 +82,52 @@ impl FiberStack {
                 rustix::mm::MprotectFlags::READ | rustix::mm::MprotectFlags::WRITE,
             )?;
 
-            Ok(Self {
+            Ok(Self::Mmap {
                 top: mmap.cast::<u8>().add(mmap_len),
                 len: mmap_len,
-                mmap: true,
             })
         }
     }
 
     pub unsafe fn from_raw_parts(base: *mut u8, len: usize) -> io::Result<Self> {
-        Ok(Self {
+        Ok(Self::Manual {
             top: base.add(len),
             len,
-            mmap: false,
         })
     }
 
+    pub fn from_custom(custom: Box<dyn RuntimeFiberStack>) -> io::Result<Self> {
+        Ok(Self::Custom(custom))
+    }
+
     pub fn top(&self) -> Option<*mut u8> {
-        Some(self.top)
+        Some(match self {
+            FiberStack::Mmap { top, len: _ } => *top,
+            FiberStack::Manual { top, len: _ } => *top,
+            FiberStack::Custom(r) => r.top(),
+        })
     }
 
     pub fn range(&self) -> Option<Range<usize>> {
-        let base = unsafe { self.top.sub(self.len) as usize };
-        Some(base..base + self.len)
+        Some(match self {
+            FiberStack::Mmap { top, len } => {
+                let base = unsafe { top.sub(*len) as usize };
+                base..base + len
+            }
+            FiberStack::Manual { top, len } => {
+                let base = unsafe { top.sub(*len) as usize };
+                base..base + len
+            }
+            FiberStack::Custom(r) => r.range(),
```

Comment:
And here for both start and end parts of the range

Reply:
Done.